### PR TITLE
fix: observe GRPC client

### PR DESCRIPTION
**Pull request by gatici**
This PR aims to solve GPRC client connection issues which appears when GPRC server (webconsole) is restarted.

If the Webconsole is restarted, other modules loses the GRPC connection to Webconsole
They tries to connect again for hours with no success.

2024-09-26T11:32:10.014Z [smf] 2024-09-26T11:32:10Z [ERRO][Config5g][GRPC] Connectivity status idle, trying to connect again

NF observes the GRPC client availability and connection status in an infinite loop.
If the connection is ready it does nothing. If GRPC client does not exist, it creates new client. If GRPC client connection is not ready, it closes the existing client and creates another client.
Only one GRPC client exists for any time slot.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -81,3 +81,22 @@ docker-push:
...
+			-coverprofile=.coverage/coverage-unit.txt \
+			-covermode=atomic \
+			-v \
+			./ ./...
+
+fmt:
+	@go fmt ./...
+
+golint:
+	@docker run --rm -v $(CURDIR):/app -w /app golangci/golangci-lint:latest golangci-lint run -v --config /app/.golangci.yml
```
Comment (gab-arrobo) [inline Makefile:102]:
```suggestion
	@docker run --rm -v $(CURDIR):/app -w /app golangci/golangci-lint:latest golangci-lint run -v --config /app/.golangci.yml

```

**Review by gab-arrobo — COMMENTED** [review]
I see that changes are not consistent across the different NFs (in some NFs the `updateConfig` is in `factory/factory.go` like this in NF/PR and in other NFs, the same function goes in `service/init.go`). Would this be a good time to make the change consistent across all NFs?

File: Makefile (hunk 1 above)
Comment (gatici) [reply Makefile]: Done

File: factory/factory.go [hunk 2]
```diff
@@ -43,14 +42,11 @@ func InitConfigFactory(f string) error {
...
 			logger.CfgLog.Infoln("MANAGED_BY_CONFIG_POD is true")
-			commChannel := client.ConfigWatcher(NssfConfig.Configuration.WebuiUri)
-			go NssfConfig.updateConfig(commChannel)
 		} else {
 			go func() {
-				logger.CfgLog.Infoln("Use helm chart config ")
+				logger.CfgLog.Infoln("use helm chart config")
 				ConfigPodTrigger <- true
 			}()
 		}
```
Comment (gab-arrobo) [inline factory/factory.go:52]: I think this `if` statement has to be move into the `service/init.go` (similarly as it was done in the `amf` https://github.com/omec-project/amf/pull/330/files#diff-3289814d5eb2b5ebf4f9002fc41f602d2af395ca60676275f6113c0defa8f485)

**Review by gab-arrobo — CHANGES_REQUESTED** [review]

File: factory/factory.go (hunk 2 above)
Comment (gatici) [reply factory/factory.go]: Done

**Comment by gab-arrobo** [comment]
@gatici, did you have a chance to run E2E tests with this PR? I am asking this because only `profile1` passes the test and all other profiles fail as you can see below:
```
        "Profile Name: profile1 , Profile Type: register",
        "Ue's Passed: 5 , Ue's Failed: 0",
        "Profile Status: PASS",
        "Profile Name: profile3 , Profile Type: anrelease",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007523, profile timeout",
        "imsi:imsi-208930100007524, profile timeout",
        "imsi:imsi-208930100007521, profile timeout",
        "imsi:imsi-208930100007520, profile timeout",
        "imsi:imsi-208930100007522, profile timeout",
        "Profile Status: FAIL",
        "Profile Name: profile7 , Profile Type: uereqpdusessrelease",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007563, profile timeout",
        "imsi:imsi-208930100007564, profile timeout",
        "imsi:imsi-208930100007560, profile timeout",
        "imsi:imsi-208930100007561, profile timeout",
        "imsi:imsi-208930100007562, profile timeout",
        "Profile Status: FAIL",
        "Profile Name: profile5 , Profile Type: deregister",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007541, profile timeout",
        "imsi:imsi-208930100007542, profile timeout",
        "imsi:imsi-208930100007543, profile timeout",
        "imsi:imsi-208930100007540, profile timeout",
        "imsi:imsi-208930100007544, profile timeout",
        "Profile Status: FAIL",
        "Profile Name: profile2 , Profile Type: pdusessest",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007513, profile timeout",
        "imsi:imsi-208930100007511, profile timeout",
        "imsi:imsi-208930100007514, profile timeout",
        "imsi:imsi-208930100007512, profile timeout",
        "imsi:imsi-208930100007510, profile timeout",
        "Profile Status: FAIL",
        "Profile Name: profile4 , Profile Type: uetriggservicereq",
        "Ue's Passed: 0 , Ue's Failed: 5",
        "Profile Errors:",
        "imsi:imsi-208930100007530, profile timeout",
        "imsi:imsi-208930100007531, profile timeout",
        "imsi:imsi-208930100007533, profile timeout",
        "imsi:imsi-208930100007532, profile timeout",
        "imsi:imsi-208930100007534, profile timeout",
        "Profile Status: FAIL"
```

**Comment by gatici** [comment]
> @gatici, did you have a chance to run E2E tests with this PR? I am asking this because only `profile1` passes the test and all other profiles fail as you can see below:
> 
> ```
>         "Profile Name: profile1 , Profile Type: register",
>         "Ue's Passed: 5 , Ue's Failed: 0",
>         "Profile Status: PASS",
>         "Profile Name: profile3 , Profile Type: anrelease",
>         "Ue's Passed: 0 , Ue's Failed: 5",
>         "Profile Errors:",
>         "imsi:imsi-208930100007523, profile timeout",
>         "imsi:imsi-208930100007524, profile timeout",
>         "imsi:imsi-208930100007521, profile timeout",
>         "imsi:imsi-208930100007520, profile timeout",
>         "imsi:imsi-208930100007522, profile timeout",
>         "Profile Status: FAIL",
>         "Profile Name: profile7 , Profile Type: uereqpdusessrelease",
>         "Ue's Passed: 0 , Ue's Failed: 5",
>         "Profile Errors:",
>         "imsi:imsi-208930100007563, profile timeout",
>         "imsi:imsi-208930100007564, profile timeout",
>         "imsi:imsi-208930100007560, profile timeout",
>         "imsi:imsi-208930100007561, profile timeout",
>         "imsi:imsi-208930100007562, profile timeout",
>         "Profile Status: FAIL",
>         "Profile Name: profile5 , Profile Type: deregister",
>         "Ue's Passed: 0 , Ue's Failed: 5",
>         "Profile Errors:",
>         "imsi:imsi-208930100007541, profile timeout",
>         "imsi:imsi-208930100007542, profile timeout",
>         "imsi:imsi-208930100007543, profile timeout",
>         "imsi:imsi-208930100007540, profile timeout",
>         "imsi:imsi-208930100007544, profile timeout",
>         "Profile Status: FAIL",
>         "Profile Name: profile2 , Profile Type: pdusessest",
>         "Ue's Passed: 0 , Ue's Failed: 5",
>         "Profile Errors:",
>         "imsi:imsi-208930100007513, profile timeout",
>         "imsi:imsi-208930100007511, profile timeout",
>         "imsi:imsi-208930100007514, profile timeout",
>         "imsi:imsi-208930100007512, profile timeout",
>         "imsi:imsi-208930100007510, profile timeout",
>         "Profile Status: FAIL",
>         "Profile Name: profile4 , Profile Type: uetriggservicereq",
>         "Ue's Passed: 0 , Ue's Failed: 5",
>         "Profile Errors:",
>         "imsi:imsi-208930100007530, profile timeout",
>         "imsi:imsi-208930100007531, profile timeout",
>         "imsi:imsi-208930100007533, profile timeout",
>         "imsi:imsi-208930100007532, profile timeout",
>         "imsi:imsi-208930100007534, profile timeout",
>         "Profile Status: FAIL"
> ```

Yes tested. It is all passing.

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**